wallet/src/main/java/com/android/identity_credential/wallet/presentation/TransferHelper.kt

```diff
@@ -0,0 +1,228 @@
+package com.android.identity_credential.wallet.presentation
+
+import android.content.Context
+import android.content.SharedPreferences
+import android.preference.PreferenceManager
+import com.android.identity.android.mdoc.deviceretrieval.DeviceRetrievalHelper
+import com.android.identity.cbor.Cbor
+import com.android.identity.credential.CredentialRequest
+import com.android.identity.credential.CredentialStore
+import com.android.identity.credential.NameSpacedData
+import com.android.identity.crypto.Algorithm
+import com.android.identity.issuance.CredentialExtensions.credentialConfiguration
+import com.android.identity.issuance.CredentialExtensions.issuingAuthorityIdentifier
+import com.android.identity.issuance.CredentialPresentationFormat
+import com.android.identity.issuance.IssuingAuthorityRepository
+import com.android.identity.mdoc.mso.MobileSecurityObjectParser
+import com.android.identity.mdoc.mso.StaticAuthDataParser
+import com.android.identity.mdoc.request.DeviceRequestParser
+import com.android.identity.mdoc.response.DeviceResponseGenerator
+import com.android.identity.mdoc.response.DocumentGenerator
+import com.android.identity.mdoc.util.MdocUtil
+import com.android.identity.util.Constants
+import com.android.identity.util.Timestamp
+import com.android.identity_credential.wallet.SelfSignedMdlIssuingAuthority
+import com.android.identity_credential.wallet.WalletApplication
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+import java.util.OptionalLong
+
+/**
+ * Transfer Helper provides helper functions for starting to process a presentation request, as well
+ * as finishing processing the request to get response bytes to send to requesting party (after user
+ * has accepted one or more dialog prompts)
+ */
+class TransferHelper(
+    private val credentialStore: CredentialStore,
+    private val issuingAuthorityRepository: IssuingAuthorityRepository,
+    private val context: Context,
+    private val deviceRetrievalHelper: DeviceRetrievalHelper,
+    private val onError: (String) -> Unit
```

Comment:
I would use a `Throwable` instead of `String` here, for better logging/handling in the callees.